### PR TITLE
Saving exercises to file

### PR DESCRIPTION
**Pull request by craig-a-lee**
when the app is oopened, it first checks to see if a file exists with the data on all exercises. If such a file does exist, then this file is used to populate the array of all exercises. If not, then a request is made using the api to get all exercises. This adds complexity and limits api usage.

### CODE REVIEW

File: Wellthier/Managers/ExerciseSharedManager.m [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+- (void) fetchAllExercisesFromApi {
+    ExerciseAPIManager *manager = [ExerciseAPIManager new];
+    [manager fetchAllExercises:^(NSArray *exercises, NSError *error) {
+        self.allExercises = exercises;
+    }];
+}
+
+- (void)fetchAllExercisesFromFile {
+//    NSDictionary *dictionary = [NSDictionary dictionaryWithContentsOfFile:[[NSBundle mainBundle] pathForResource:@"Values" ofType:@"plist"]];
```
Comment (vijaykanse) [inline Wellthier/Managers/ExerciseSharedManager.m]: remove

File: Wellthier/Managers/ExerciseSharedManager.m [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+
+    //NSString *path = @"/Users/craiglee/Desktop/Projects/Wellthier/Wellthier/ExerciseData.plist";
```
Comment (vijaykanse) [inline Wellthier/Managers/ExerciseSharedManager.m]: remove

File: Wellthier/Managers/ExerciseSharedManager.m [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    NSDictionary *theDict = [NSDictionary dictionaryWithContentsOfFile:@"ExerciseData.plist"];
+    // NSFileManager *manager = [NSFileManager defaultManager];
```
Comment (vijaykanse) [inline Wellthier/Managers/ExerciseSharedManager.m]: remove

File: Wellthier/DocManagers/ExerciseDoc.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//  Wellthier
+//
+//  Created by Craig Lee on 8/1/22.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface ExerciseDoc : NSObject
```
Comment (vijaykanse) [inline Wellthier/DocManagers/ExerciseDoc.h]: We need to think better naming convention

**Review by xueyike — APPROVED** [review]
What a giant diff! Next time maybe you could split the diff with 2 section, one for library import and one for your own code changes.

File: Wellthier/DocManagers/ExerciseDoc.h (hunk 4 above)
Comment (xueyike) [reply Wellthier/DocManagers/ExerciseDoc.h]: Something like ExerciseDatabase?

**Comment by craig-a-lee** [comment]
> What a giant diff! Next time maybe you could split the diff with 2 section, one for library import and one for your own code changes.

i dont think i will be using the library import anymore so will remove it

**Pull request merged**